ts/components/ui/TabItem.tsx

```diff
@@ -0,0 +1,190 @@
+import React from "react";
+import { GestureResponderEvent, Pressable, StyleSheet } from "react-native";
+import Animated, {
+  Extrapolate,
+  interpolate,
+  interpolateColor,
+  useAnimatedStyle,
+  useDerivedValue,
+  useSharedValue,
+  withSpring
+} from "react-native-reanimated";
+import { WithTestID } from "../../types/WithTestID";
+import { IOIcons, Icon } from "../core/icons";
+import { HSpacer } from "../core/spacer/Spacer";
+import { LabelHeader } from "../core/typography/LabelHeader";
+import { IOScaleValues, IOSpringValues } from "../core/variables/IOAnimations";
+import { IOColors } from "../core/variables/IOColors";
+
+type ColorMode = "light" | "dark";
+
+type TabItem = WithTestID<{
+  label: string;
+  color?: ColorMode;
+  selected?: boolean;
+  // Icons
+  icon?: IOIcons;
+  iconSelected?: IOIcons;
+  // Accessibility
+  accessibilityLabel: string;
+  accessibilityHint?: string;
+  // Events
+  onPress: (event: GestureResponderEvent) => void;
+}>;
+
+type ColorStates = {
+  default: {
+    background: string;
+    foreground: IOColors;
+  };
+  selected: {
+    background: string;
+    foreground: IOColors;
+  };
+  pressed: {
+    background: string;
+  };
+};
+
+const mapColorStates: Record<NonNullable<TabItem["color"]>, ColorStates> = {
```

Comment:
```suggestion
const colorStatesMap: Record<NonNullable<TabItem["color"]>, ColorStates> = {
```
since we are talking about a `map` dataType instead of a `mapFunc`, I believe this naming would be more fitting; what do you think?